### PR TITLE
local opsworks cluster.

### PR DESCRIPTION
**Pull request by djcp**

### CODE REVIEW

File: README.local-opsworks.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+        ./bin/local_cluster up
+
+* Access your environment by pointing your web browser to http://10.10.10.10/
+  The default password is "fakepass".
+
+## `./bin/local_cluster`
+
+This harness helps you spin up a multi-machine cluster. Run it without
+arguments to see what it supports. It ultimately use the `vagrant` command
```
Comment (lbjay) [inline README.local-opsworks.md]: "uses"

File: README.local-opsworks.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        /vagrant/mh-opsworks-recipes
+
+* Do work locally. After you've done stuff, redeploy your changes. This takes
+  around 5 minutes (for now).
+
+        ./bin/local_cluster deploy
+
+* Stop your cluster
+
+        ./bin/local_cluster halt
```
Comment (karendolan) [inline README.local-opsworks.md:99]: Dan, If I halt my cluster, is that the same as vagrant suspend? How do I restart and existing halted cluster?

File: README.local-opsworks.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        vagrant plugin install vagrant-hosts --plugin-version 2.6.2
+
+* Ensure your repository directory structure looks like this:
+
+        [parent dir]
+          - mh-opsworks
+          - mh-opsworks-recipes
+          - matterhorn-dce-fork
+
+* Start your environment! This takes about 20 to 25 minutes on my beefy machine
```
Comment (karendolan) [inline README.local-opsworks.md:32]: Dan, would I run this after a ./bin/local_cluster halt? Would that still take 20-25 min to restart the cluster?

File: README.local-opsworks.md (hunk 2 above)
Comment (djcp) [inline README.local-opsworks.md:99]: No, "halt" stops the machines completely. A subsequent "./bin/local_cluster up" takes about 6 minutes. I'll implement a "suspend" too - good idea!  You could always do the suspend manually, of course - "vagrant suspend" would work fine.

The only reason I wrote `local_cluster` is to speed up cluster starts and deploys - by default, vagrant starts each instance in your Vagrantfile in order, when it's usually fine to start more than one at a time.

Comment (djcp) [inline README.local-opsworks.md:99]: sadly, "vagrant resume" does a provision by default, and it shouldn't.  https://github.com/mitchellh/vagrant/issues/6787 I'll work through the tooling, though.

File: README.local-opsworks.md (hunk 3 above)
Comment (djcp) [inline README.local-opsworks.md:32]: Yes, "up" and "halt" are the opposites of each other in vagrant-land. It takes about 6 minutes on my machine to "up" a cluster that's previously been halted. 

I'm working on suspend / resume tooling and that should speed things up too.

Comment (djcp) [inline README.local-opsworks.md:32]: Ok - resuming and suspending a previously provisioned cluster takes around 50 seconds each way now that I've parallelized it.

File: README.local-opsworks.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  with SSD.
+
+        ./bin/local_cluster up
+
+* Access your environment by pointing your web browser to http://10.10.10.10/
+  The default password is "fakepass".
+
+## `./bin/local_cluster`
```
Comment (lbjay) [inline README.local-opsworks.md:48]: pretty minor but the markdown rendering makes this look strange, like it's the last item of the previous unordered list. Maybe change to `## local_cluster harness`?

Comment (lbjay) [inline README.local-opsworks.md:48]: also the initial run w/o arguments does the box install which i wasn't expecting. no biggie.

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,183 @@
+require 'rbconfig'
+
+Vagrant.require_version(">= 1.7.0")
```
Comment (lbjay) [inline Vagrantfile:3]: readme indicates v1.8.x required

Comment (djcp) [inline Vagrantfile:3]: Yeah, I inherited this from the upstream "opsworks-vm" project. 1.7 would probably work but I've only tested with 1.8

**Pull request merged**